lib/truffle/socket/truffle.rb

```diff
@@ -95,10 +95,18 @@ def self.accept(source, new_class)
       end
     end
 
-    def self.accept_nonblock(source, new_class)
+    def self.accept_nonblock(source, new_class, exception)
       source.fcntl(::Fcntl::F_SETFL, ::Fcntl::O_NONBLOCK)
 
-      accept(source, new_class)
+      begin
+        accept(source, new_class)
+      rescue Errno::EAGAIN
```

Comment:
`accept` should probably throw a IO::WaitReadable exception and not just EAGAIN, but that's out of scope.